components/background_hang_monitor/tests/hang_monitor_tests.rs

```diff
@@ -117,3 +105,30 @@ fn test_hang_monitoring() {
     // Still no new alerts because the hang monitor has shut-down already.
     assert!(background_hang_monitor_receiver.try_recv().is_err());
 }
+
+#[test]
+fn test_hang_monitoring_unregister() {
+    let (background_hang_monitor_ipc_sender, background_hang_monitor_receiver) =
+        ipc::channel().expect("ipc channel failure");
+
+    let background_hang_monitor_register =
+        HangMonitorRegister::init(background_hang_monitor_ipc_sender.clone());
+    let background_hang_monitor = background_hang_monitor_register.register_component(
+        MonitoredComponentId(TEST_PIPELINE_ID, MonitoredComponentType::Script),
+        Duration::from_millis(10),
+        Duration::from_millis(1000),
+    );
+
+    // Start an activity.
+    let hang_annotation = HangAnnotation::Script(ScriptHangAnnotation::AttachLayout);
+    background_hang_monitor.notify_activity(hang_annotation);
+
+    // Sleep until the "transient" timeout has been reached.
+    thread::sleep(Duration::from_millis(10));
+
+    // Unregister the component.
+    background_hang_monitor.unregister();
```

Comment:
Let's put the `sleep` after the call to unregister, I'm afraid that this might fail intermittently if the monitor sends an alert just before the call to `unregister` comes through.